presto-hive/src/main/java/com/facebook/presto/hive/HiveWriterFactory.java

```diff
@@ -334,23 +339,38 @@ public HiveWriter createWriter(Page partitionColumns, int position, OptionalInt
             }
         }
         else {
-            // Write to: an existing partition in an existing partitioned table,
-            if (bucketNumber.isPresent()) {
-                throw new PrestoException(HIVE_PARTITION_READ_ONLY, "Cannot insert into existing partition of bucketed Hive table: " + partitionName.get());
-            }
-            if (immutablePartitions) {
-                throw new PrestoException(HIVE_PARTITION_READ_ONLY, "Cannot insert into an existing partition of Hive table: " + partitionName.get());
+            // Write to: an existing partition in an existing partitioned table
+            if (!overwriteExistingPartitions) {
+                // Append to an existing partition
+                if (bucketNumber.isPresent()) {
+                    throw new PrestoException(HIVE_PARTITION_READ_ONLY, "Cannot append into existing partition of bucketed Hive table: " + partitionName.get());
+                }
+                if (immutablePartitions) {
+                    throw new PrestoException(HIVE_PARTITION_READ_ONLY, "Cannot append into an existing partition of Hive table: " + partitionName.get());
+                }
+                updateMode = APPEND;
+                checkTableAndPartitionSchemaMatch(table, tableName, partition.get(), partitionName.get());
+
+                HiveWriteUtils.checkPartitionIsWritable(partitionName.get(), partition.get());
+
+                outputStorageFormat = partition.get().getStorage().getStorageFormat();
+                schema = getHiveSchema(partition.get(), table);
+
+                writeInfo = locationService.getPartitionWriteInfo(locationHandle, partition, partitionName.get());
             }
-            updateMode = APPEND;
-            checkTableAndPartitionSchemaMatch(table, tableName, partition.get(), partitionName.get());
+            else {
+                // Overwrite an existing partition
+                updateMode = OVERWRITE;
+                HiveWriteUtils.checkTableAndPartitionSchemaMatch(table, tableName, partition.get(), partitionName.get());
 
-            // Append to an existing partition
-            HiveWriteUtils.checkPartitionIsWritable(partitionName.get(), partition.get());
+                HiveWriteUtils.checkPartitionIsWritable(partitionName.get(), partition.get());
```

Comment:
This check also needs to be deleted or adjusted. See what `metadataDelete` does.